src/java/org/apache/cassandra/db/partitions/BTreePartitionUpdater.java

```diff
@@ -149,12 +146,11 @@ public boolean abortEarly()
 
     public Cell<?> merge(Cell<?> previous, Cell<?> insert)
     {
-        if (insert != previous)
-        {
-            long timeDelta = Math.abs(insert.timestamp() - previous.timestamp());
-            if (timeDelta < colUpdateTimeDelta)
-                colUpdateTimeDelta = timeDelta;
-        }
+        if (insert == previous)
```

Comment:
previously on ds-trunk we would return a cloned instance of `insert`. Now we return the original insert.
We clone insert only if it is not equal to previous.
Why is cloning needed?

Reply:
I am not sure - maybe when `insert == previous` this means that it was already inserted and `previous` itself is a clone?

btw, this was added to C* 4.0 with CASSANDRA-18125, although there is it located in `AtomicBTreeParition.merge`.